xxxx-release-and-versioning.md

```diff
@@ -27,45 +37,51 @@ Only a the latest `MAJOR.MINOR.PATCH` version of the last 2 `MAJOR.MINOR` versio
-                     |         |    |         |
-                     v         v    v         v
- stable -------------o---------o----o---------o-------------------->
-                  {a1,a2} \   {a3} {b1,b2}   {c1,c2,c3} \
-                           \    \                        \
-                            \    +------------            \
-                             \                \            \
- stable-1.0                   t----------o-----t------------------->
-                         v1.0.0 {a1,a2}     v1.0.1 {a3}      \
-                                                              \
- stable-1.1                                                    t--->
-                                                             v1.1.0
+feature2    -o---------o----------o
+                                   \
+                                    \
+feature1    -o                       \
+              \                       \
+               v                       v
+master      ----o-----------------------o-------------------------------->
+                             \                           \
+                              v                           \
+stable-1.0                     o-----------o----o----------^------------->
+                            v1.0.0           v1.0.1         \
+                                                             v
+stable-1.1                                                    o---------->
+                                                            v1.1.0
 ```
 
-In this chart, changes are selectively backported from `develop` to `stable` in a per-feature basis (`{a1,a2}`, `{b1,b2}`, ...) This process can be done by cherry-picking changes and fixing conflicts as necessary.
+#### Feature Branches
+
+Feature branches, such as `feature2` and `feature1`, are development branches that implements new feature or change existing functionality. These branches may be long-running and should be merged to `master` only when the feature is considered stable enough for public consumption. Feature branches should regularly sync up with master to avoid too much code diversion. Feature branches may have its own public testing environment for each branch with its data copied from staging on each deploy.
+
+#### `master` branch
+
+The `master` branch is the branch that development release will be based on. While we required that feature branch must be stable enough before it could be merged into `master`, the `master` branch may still not stable enough to use in production environment. The `master` branch is where a stable branch will be branched from when it is considered stable enough. Before branching into stable, `master` may undergo a period of feature freeze, in which new features are not allowed to be merged onto the master branch, and only stability or bugfixes are accepted. The master branch will be deployed automatically to the OmiseGO staging environment for internal and public testing.
```

Comment:
While we require that feature branches must be stable enough before it getting merged into `master`, the `master` branch may still not be stable enough to use in a production environment.

Reply:
My sense is that people working in production would understand that `master` is latest and greatest and versioned releases are for production. At least this is what I see in the ruby/rails/js/node worlds.